client/client.go

```diff
@@ -5,18 +5,20 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"net/http"
 
 	apps "github.com/replicatedhq/replicated/gen/go/apps"
 	channels "github.com/replicatedhq/replicated/gen/go/channels"
 	releases "github.com/replicatedhq/replicated/gen/go/releases"
+	v2 "github.com/replicatedhq/replicated/gen/go/v2"
 )
 
 const apiOrigin = "https://api.replicated.com/vendor"
 
 // Client provides methods to manage apps, channels, and releases.
 type Client interface {
-	GetAppBySlug(slug string) (*apps.App, error)
+	GetApp(slugOrID string) (*apps.App, error)
 	CreateApp(name string) (*apps.App, error)
```

Comment:
i kinda prefer something like this for forwards compatibility. Probably applies to other function signatures as well

```go
CreateApp(opts CreateAppOptions) (*apps.App, error)
```